### PR TITLE
vkd3d: Refactor platform-specific code

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
This PR moves the module handling code which was previously dumped in device.c and the code to retrieve the current executable path to its own file.

This also eliminates HAVE_DECL_PROGRAM_INVOCATION_NAME from config.h

It also implements vkd3d_get_program_name for Win32.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -110,6 +110,7 @@ libvkd3d_la_SOURCES = \
 	libs/vkd3d/command.c \
 	libs/vkd3d/device.c \
 	libs/vkd3d/meta.c \
+	libs/vkd3d/platform.c \
```
Comment (HansKristian-Work) [inline Makefile.am:113]: New headers should also be added to Makefile.am

File: libs/vkd3d/platform.c [hunk 2]
```diff
@@ -0,0 +1,137 @@
+#include "vkd3d_platform.h"
```
Comment (HansKristian-Work) [inline libs/vkd3d/platform.c:1]: Need copyright header.

File: Makefile.am (hunk 1 above)
Comment (Joshua-Ashton) [reply Makefile.am:113]: Wack. Okay.

Comment (Joshua-Ashton) [reply Makefile.am:113]: We're missing the one for vkd3d_atomic.h in that case...

Not sure why the headers are included in the autotools sources given they're just going to be ignored but... I'll add another commit to make this consistent anyway

Comment (Joshua-Ashton) [reply Makefile.am:115]: Done.

File: libs/vkd3d/platform.c (hunk 2 above)
Comment (Joshua-Ashton) [reply libs/vkd3d/platform.c:19]: Done.

File: Makefile.am (hunk 1 above)
Comment (HansKristian-Work) [reply Makefile.am:115]: Ye, not sure either. Just following convention. It makes sense if there is some magical IDE tooling with automake (which I doubt).

**Pull request merged**